### ISSUE TITLE
grunt-usemin is DEPRECATED :(

### ISSUE THREAD

**Issue by stevemao** [yeoman/grunt-usemin#514]
I teared up when I saw this and instinctively tried to find an alternative (a even better tool I was hoping). I searched all the opening issues but found none related to the deprecation. I'm not sure if we should continue using any other tool to do the same job or we should go back to configure `cssmin`, `uglify`, `concat` and `filerev` manually. 

A lot of Yeoman generators are still using grunt-usmin and I'm sure I will feel nostalgic every time I open my Gruntfile.js.

**Comment by sindresorhus** [comment]
It's still useable, the problem is that nobody is interested in maintaining it. We've tried for more than a year to get a maintainer, but the issues and PRs just keeps piling up...

https://github.com/yeoman/grunt-usemin/issues/313

**Comment by stephanebachelier** [comment]
@sindresorhus I'm not very active for the moment but I plan to put a little more energy on this repo. It's the central piece for lots of my works and there are no known piece of software that does what usemin do.

**Comment by eddiemonge** [comment]
Removed the deprecated notice but if its still orphaned in a few months Ill be sad. Of course I could spend more time helping on it.

**Comment by sindresorhus** [comment]
If nothing more has happened in a few months I *will* deprecate it for real. Not doing so would be an disservice to users.

**Comment by stephanebachelier** [comment]
@sindresorhus I have started to review issues and tagging issues with the following labels:
- docs which I've started to explain in #519 
- question for any issues that need a discussion
- codereview when the issue is waiting for a PR to be merged
- waiting for any feedback needed
- feature when this is about adding something to usemin, like switching to HTML parser (under investigation)

My feeling is that most issues will be tagged by `waiting` or `docs`.

**Comment by arthurvr** [comment]
@stephanebachelier You're on fire! :fire: Doing great man!

**Comment by stephanebachelier** [comment]
@arthurvr :)

**Comment by stevemao** [comment]
@stephanebachelier it might be easier if you merge similar issues just to make it easier to manage.

**Comment by stephanebachelier** [comment]
@stevemao you're right and I've started to add the label duplicate on some issues. 
But some of them are more variants of the same problem.

**Comment by arthurvr** [comment]
> If nothing more has happened in a few months I will deprecate it for real. Not doing so would be an disservice to users.

@sindresorhus The latest commit to the `dev` branch was April 13 so sadly I start to believe we should really do it.

Don't have anything to say about this task at all, but it's indeed a disservice and kind of a pain. I also start to believe we don't only need an active maintainer but we need a rewrite.

**Comment by stevemao** [comment]
> we need a rewrite.

:+1:

**Comment by stephanebachelier** [comment]
@arthurvr I was kept by client projects and not being able to work on open source projects. I will soon have time to work again on dev branch, and first fix my PRs.

By the way it's not so easy because there are not a lot of activity and no real code review. 
Some guys don't seem to care about being reviewed, and I can go the same way on the dev branch. At least I will be happy to continue my effort without waiting most of the time in hope for a code review.

> we need a rewrite

@stevemao it's already being done in the dev branch, with the addition of HTML parser. If somebody wants to help with some tests against some major yeoman generators it would be great.

I've even though about forking grunt-usemin and going the way I think, because as already said before all my projects depends on grunt-usemin, and due to lack of activity.

**Comment by stevemao** [comment]
@stephanebachelier I understand. Not saying I don't appreciate your work but I think a rewrite can help (better logics, better codebase etc) since you've learned from the current code.
Node has changed a lot but this project does not update quickly to catch up so a lot of approaches might not be the best now.
Just my 2c.

**Comment by stevemao** [comment]
@stephanebachelier Sorry didn't know the dev branch is the rewrite. Will try out in some projects. Thanks.

**Comment by stevemao** [comment]
@stephanebachelier I think you can propose your ideas and let people see if they are good. Since there's no active maintainers its ok to try something crazy.

**Comment by arthurvr** [comment]
> At least I will be happy to continue my effort without waiting most of the time in hope for a code review.

:+1: You can still ask us to code review the whole dev branch once we want to land it on master.

> it's already being done in the dev branch,

@stephanebachelier That's not being done there. There are big changes going on there but not the changes [we originally planned to do](https://github.com/yeoman/grunt-usemin/issues/294#issuecomment-46881715) in the rewrite.

**Comment by stephanebachelier** [comment]
@arthurvr it's the first I see this issue (closed) and requirement list about usemin rewrite!

**Comment by arthurvr** [comment]
@stephanebachelier @sindresorhus @eddiemonge So what's the long term plan? Do we want to stay developing this as-is in the dev branch? Do we want to look into finally cleaning up and doing the big rewrite we ever planned out a long time ago?

I'm not sure what the best thing to do with this task. Using the current API it's a pain to maintain (even if we're able to get rid of the regexes)

**Comment by eddiemonge** [comment]
I think we should find a way to split it up into smaller, more maintainable modules and then have this as a wrapper that pulls them in. Each of those smaller modules could be rewritten cleanly. Then we could probably eventually fade this out in favor of the other modules. 
I think @sindresorhus might agree with this approach.

**Comment by stephanebachelier** [comment]
@arthurvr @eddiemonge I agree with a complete rewrite with more maintainable modules. Much better than the current base code. I was trying to pull into a better direction with dev branch but it will be much easier to rewrite from scratch.

**Comment by frankstallone** [comment]
Someone point me to the tip jar, please!

**Comment by fatso83** [comment]
Over the last month there have been several people that has stepped up in #313 that would like to help maintain this.
